dns_server/src/crawler/mod.rs

```diff
@@ -14,127 +14,71 @@
     dns_server::ServerCommands,
     error::DnsServerError,
 };
 
-use self::storage::{DnsServerStorage, DnsServerStorageRead, DnsServerTransactionRo};
+use self::{
+    core::{CoreCallback, CoreCommand, CoreEvent, CrawlerCore, CrawlerCoreConfig},
+    storage::{DnsServerStorage, DnsServerStorageRead, DnsServerTransactionRo},
+};
 
 /// How often the server performs maintenance (tries to connect to new nodes)
 const HEARTBEAT_INTERVAL: Duration = Duration::from_secs(5);
 
-/// How many outbound connection attempts can be made per heartbeat
-const MAX_CONNECTS_PER_HEARTBEAT: usize = 25;
-
-/// When the server drops the unreachable node address. Used for negative caching.
-const PURGE_UNREACHABLE_TIME: Duration = Duration::from_secs(3600);
-
-/// When the server drops the unreachable node address that was once reachable. This should take about a month.
-/// Such a long time is useful if the server itself has prolonged connectivity problems.
-const PURGE_REACHABLE_FAIL_COUNT: u32 = 35;
-
 const STORAGE_VERSION: u32 = 1;
 
-/// Connection state of a potential node address (outbound only)
-#[derive(Debug, Clone, Copy, PartialEq, Eq)]
-enum AddressState {
-    Disconnected,
-    Connecting,
-    Connected,
-}
-
-/// Additional state of a potential node address
-struct AddressData {
-    /// Connection state
-    state: AddressState,
-
-    /// Last time when the connection state was updated
-    state_updated_at: tokio::time::Instant,
-
-    /// The number of failed connection attempts.
-    /// New connection attempts are made after a progressive backoff time.
-    /// Resets to 0 when an outbound connection to the address is successful.
-    fail_count: u32,
-
-    /// Whether the address was reachable at least once.
-    /// Addresses that were once reachable are stored in the DB.
-    was_reachable: bool,
-
-    /// Whether the address was added from the command line
-    user_added: bool,
-}
-
 #[derive(Clone)]
 pub struct CrawlerConfig {
+    /// Manually specified list of nodes to connect to
     pub add_node: Vec<String>,
 
-    pub p2p_port: u16,
+    /// Default p2p port on which nodes normally accept inbound connections
+    pub default_p2p_port: u16,
 }
 
-pub struct Crawler<N: NetworkingService, S> {
-    /// Crawler config
-    config: CrawlerConfig,
-
-    /// Chain config
-    chain_config: Arc<ChainConfig>,
-
+pub struct CrawlerCallback<N: NetworkingService, S> {
     /// Backend's ConnectivityHandle
     conn: N::ConnectivityHandle,
 
     /// Backend's SyncingMessagingHandle
     sync: N::SyncingMessagingHandle,
 
-    /// Map of all known addresses (including currently unreachable)
-    addresses: BTreeMap<N::Address, AddressData>,
-
-    /// Map of all currently connected outbound peers
-    outbound_peers: BTreeMap<N::PeerId, N::Address>,
-
     /// Storage implementation
     storage: S,
 
     /// Channel used to manage the DNS server
     command_tx: mpsc::UnboundedSender<ServerCommands>,
```

Comment:
Two things:

1. Please rename `ServerCommands` to `DNSServerCommand`
2. Please rename `command_tx` to `dns_server_cmd_tx`

Reply:
Sure, updated.